File: android-native/app/src/main/java/com/example/kpkn/screens/home/HomeProgramsSection.kt
package com.example.kpkn.screens.home

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyRow
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.kpkn.data.models.Program
import com.example.kpkn.data.models.totalProgramWeeks
import com.example.kpkn.ui.components.SectionHeader
import com.example.kpkn.ui.theme.HomeCardSurface
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add

@Composable
fun HomeProgramsSection(
    programs: List<Program>,
    activeProgramId: String?,
    onProgramClick: (String) -> Unit,
    onCreateProgram: () -> Unit,
    modifier: Modifier = Modifier,
) {
    Column(modifier.fillMaxWidth()) {
        SectionHeader("Tus Programas", Modifier.padding(horizontal = 24.dp))

        if (programs.isEmpty()) {
            EmptyProgramsCard(
                modifier = Modifier.padding(horizontal = 24.dp),
                onClick = onCreateProgram,
            )
        } else {
            LazyRow(
                contentPadding = PaddingValues(horizontal = 24.dp),
                horizontalArrangement = Arrangement.spacedBy(12.dp),
            ) {
                items(programs, key = { it.id }) { program ->
                    ProgramCard(
                        program = program,
                        isActive = program.id == activeProgramId,
                        onClick = { onProgramClick(program.id) },
                    )
                }
                item(key = "create-program") {
                    CreateProgramChip(onClick = onCreateProgram)
                }
            }
        }
    }
}

@Composable
private fun ProgramCard(program: Program, isActive: Boolean, onClick: () -> Unit) {
    val bgColors = programCardCoverColors(program.coverImage)
    Column(modifier = Modifier.width(176.dp)) {
        Card(
            onClick = onClick,
            modifier = Modifier
                .fillMaxWidth()
                .height(112.dp),
            shape = RoundedCornerShape(22.dp),
            colors = CardDefaults.cardColors(containerColor = Color.Transparent),
        ) {
            Box(Modifier.fillMaxSize()) {
                Box(
                    Modifier
                        .fillMaxSize()
                        .background(Brush.linearGradient(bgColors)),
                )
                Box(
                    Modifier
                        .fillMaxSize()
                        .background(Color.Black.copy(alpha = 0.38f)),
                )
                Column(
                    modifier = Modifier
                        .fillMaxSize()
                        .padding(14.dp),
                    verticalArrangement = Arrangement.SpaceBetween,
                ) {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.Top,
                    ) {
                        if (isActive) {
                            Surface(
                                shape = RoundedCornerShape(999.dp),
                                color = MaterialTheme.colorScheme.primary.copy(alpha = 0.92f),
                                contentColor = MaterialTheme.colorScheme.onPrimary,
                            ) {
                                Text(
                                    "ACTIVO",
                                    modifier = Modifier.padding(horizontal = 7.dp, vertical = 3.dp),
                                    style = MaterialTheme.typography.labelSmall,
                                    fontWeight = FontWeight.Black,
                                    fontSize = 9.sp,
                                )
                            }
                        }
                        Text(
                            when (program.mode) {
                                com.example.kpkn.data.models.ProgramMode.POWERLIFTING -> "Powerlifting"
                                com.example.kpkn.data.models.ProgramMode.POWERBUILDING -> "Powerbuilding"
                                com.example.kpkn.data.models.ProgramMode.HYPERTROPHY -> "Hipertrofia"
                            },
                            style = MaterialTheme.typography.labelSmall,
                            color = Color.White.copy(alpha = 0.64f),
                            maxLines = 1,
                            fontSize = 9.sp,
                        )
                    }
                    Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
                        Text(
                            program.name,
                            style = MaterialTheme.typography.titleSmall,
                            fontWeight = FontWeight.Black,
                            color = Color.White,
                            maxLines = 2,
                            overflow = TextOverflow.Ellipsis,
                        )
                        Text(
                            "${program.totalProgramWeeks.coerceAtLeast(1)} semanas",
                            style = MaterialTheme.typography.labelSmall,
                            color = Color.White.copy(alpha = 0.62f),
                        )
                    }
                }
            }
        }
    }
}

private fun programCardCoverColors(coverValue: String?): List<Color> = when (coverValue) {
    "gradient://lagoon" -> listOf(Color(0xFF0D1B2A), Color(0xFF1B4965), Color(0xFF5FA8D3))
    "gradient://velvet" -> listOf(Color(0xFF1C1024), Color(0xFF5B2A86), Color(0xFFE26D5A))
    "gradient://forest" -> listOf(Color(0xFF102A1F), Color(0xFF2D6A4F), Color(0xFF95D5B2))
    else -> listOf(Color(0xFF20110F), Color(0xFF8D3D2E), Color(0xFFE08E45))
}


@Composable
private fun CreateProgramChip(onClick: () -> Unit) {
    Card(
        onClick = onClick,
        modifier = Modifier
            .width(140.dp)
            .height(112.dp),
        shape = RoundedCornerShape(22.dp),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.surfaceVariant.copy(alpha = 0.55f),
        ),
    ) {
        Column(
            Modifier.fillMaxSize().padding(14.dp),
            verticalArrangement = Arrangement.Center,
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            Icon(Icons.Default.Add, contentDescription = null, tint = MaterialTheme.colorScheme.primary)
            Spacer(Modifier.height(8.dp))
            Text(
                "Nuevo programa",
                style = MaterialTheme.typography.labelMedium,
                fontWeight = FontWeight.Bold,
                color = MaterialTheme.colorScheme.onSurface,
            )
        }
    }
}

@Composable
private fun EmptyProgramsCard(modifier: Modifier = Modifier, onClick: () -> Unit) {
    Card(
        onClick = onClick,
        modifier = modifier.fillMaxWidth(),
        shape = RoundedCornerShape(22.dp),
        colors = CardDefaults.cardColors(containerColor = HomeCardSurface),
    ) {
        Box(
            modifier = Modifier.fillMaxWidth().padding(28.dp),
            contentAlignment = Alignment.Center,
        ) {
            Column(horizontalAlignment = Alignment.CenterHorizontally) {
                Text(
                    "Crea tu primer programa",
                    style = MaterialTheme.typography.bodyMedium,
                    color = Color.White,
                    fontWeight = FontWeight.Bold,
                )
                Spacer(Modifier.height(4.dp))
                Text(
                    "La Home se llenará con tus sesiones y métricas cuando lo actives.",
                    style = MaterialTheme.typography.bodySmall,
                    color = Color.White.copy(alpha = 0.5f),
                )
            }
        }
    }
}

File: ios-native/KPKNFit/KPKNFit/Presentation/Components/HomeProgramsSection.swift
import SwiftUI

private let CardDark = Color(hex: 0x1C1C1E)

/// Swift translation of HomeProgramsSection.kt
public struct HomeProgramsSection: View {
    let programs: [Program]
    let activeProgramId: String?
    let onProgramClick: (String) -> Void
    let onCreateProgram: () -> Void
    
    public init(
        programs: [Program],
        activeProgramId: String?,
        onProgramClick: @escaping (String) -> Void,
        onCreateProgram: @escaping () -> Void
    ) {
        self.programs = programs
        self.activeProgramId = activeProgramId
        self.onProgramClick = onProgramClick
        self.onCreateProgram = onCreateProgram
    }
    
    public var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            SectionHeader("Tus Programas")
                .padding(.horizontal, 24)
            
            if programs.isEmpty {
                EmptyProgramsCard(onClick: onCreateProgram)
                    .padding(.horizontal, 24)
            } else {
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: 12) {
                        ForEach(programs) { program in
                            ProgramCard(
                                program: program,
                                isActive: program.id == activeProgramId,
                                onClick: { onProgramClick(program.id) }
                            )
                        }
                    }
                    .padding(.horizontal, 24)
                }
            }
        }
    }
}

// ─── Program Card ────────────────────────────────────────────────────────────

private struct ProgramCard: View {
    let program: Program
    let isActive: Bool
    let onClick: () -> Void
    
    var body: some View {
        Button(action: onClick) {
            VStack(alignment: .leading, spacing: 0) {
                ZStack {
                    // Linear Gradient cover background
                    LinearGradient(
                        colors: programCardCoverColors(program.coverImage),
                        startPoint: .topLeading,
                        endPoint: .bottomTrailing
                    )
                    
                    // Dark overlay (0.38)
                    Color.black.opacity(0.38)
                    
                    // Card content
                    VStack(alignment: .leading, spacing: 0) {
                        // Top Row: Active Pill + Mode
                        HStack(alignment: .top) {
                            if isActive {
                                Text("ACTIVO")
                                    .font(.system(size: 9, weight: .black))
                                    .foregroundColor(.black)
                                    .padding(.horizontal, 7)
                                    .padding(.vertical, 3)
                                    .background(AppColors.neonYellow.opacity(0.92)) // primary color equivalent
                                    .cornerRadius(999)
                            }
                            
                            Spacer()
                            
                            Text(programModeLabel(program.mode))
                                .font(.system(size: 9))
                                .foregroundColor(Color.white.opacity(0.64))
                                .lineLimit(1)
                        }
                        
                        Spacer()
                        
                        // Bottom Column: Name + Weeks
                        VStack(alignment: .leading, spacing: 2) {
                            Text(program.name)
                                .font(.system(size: 14, weight: .black))
                                .foregroundColor(.white)
                                .lineLimit(2)
                                .multilineTextAlignment(.leading)
                            
                            Text("\(max(program.totalProgramWeeks, 1)) semanas")
                                .font(.system(size: 10))
                                .foregroundColor(Color.white.opacity(0.62))
                        }
                    }
                    .padding(14)
                }
                .frame(width: 176, height: 112)
                .cornerRadius(22)
            }
        }
        .buttonStyle(PlainButtonStyle())
    }
}

// Helper to determine cover colors
private func programCardCoverColors(_ coverValue: String?) -> [Color] {
    switch coverValue {
    case "gradient://lagoon":
        return [Color(hex: 0x0D1B2A), Color(hex: 0x1B4965), Color(hex: 0x5FA8D3)]
    case "gradient://velvet":
        return [Color(hex: 0x1C1024), Color(hex: 0x5B2A86), Color(hex: 0x5FA8D3)] // Note Velvet has different ending in original
    case "gradient://forest":
        return [Color(hex: 0x102A1F), Color(hex: 0x2D6A4F), Color(hex: 0x95D5B2)]
    default:
        return [Color(hex: 0x20110F), Color(hex: 0x8D3D2E), Color(hex: 0xE08E45)]
    }
}

private func programModeLabel(_ mode: ProgramMode) -> String {
    switch mode {
    case .POWERLIFTING: return "Powerlifting"
    case .POWERBUILDING: return "Powerbuilding"
    case .HYPERTROPHY: return "Hipertrofia"
    }
}

// ─── Empty Programs Card ──────────────────────────────────────────────────────

private struct EmptyProgramsCard: View {
    let onClick: () -> Void
    
    var body: some View {
        Button(action: onClick) {
            VStack(alignment: .center, spacing: 4) {
                Text("Crea tu primer programa")
                    .font(.body)
                    .fontWeight(.bold)
                    .foregroundColor(.white)
                
                Text("La Home se llenará con tus sesiones y métricas cuando lo actives.")
                    .font(.caption)
                    .foregroundColor(Color.white.opacity(0.5))
                    .multilineTextAlignment(.center)
            }
            .padding(.vertical, 28)
            .padding(.horizontal, 24)
            .frame(maxWidth: .infinity, alignment: .center)
            .background(CardDark)
            .cornerRadius(22)
        }
        .buttonStyle(PlainButtonStyle())
    }
}

#Preview {
    ZStack {
        Color.black.ignoresSafeArea()
        VStack(spacing: 20) {
            HomeProgramsSection(
                programs: [
                    Program(id: "1", name: "Ripped to Shreds", coverImage: "gradient://lagoon", mode: .HYPERTROPHY),
                    Program(id: "2", name: "Max Power 5x5", coverImage: "gradient://velvet", mode: .POWERLIFTING)
                ],
                activeProgramId: "1",
                onProgramClick: { _ in },
                onCreateProgram: {}
            )
            
            HomeProgramsSection(
                programs: [],
                activeProgramId: nil,
                onProgramClick: { _ in },
                onCreateProgram: {}
            )
        }
    }
}
